packages/react-components/react-dialog/Spec.md

```diff
@@ -1,63 +1,524 @@
+    </DialogBody>
+    <DialogActions>
+      <DialogTrigger type="close">
+        <Button>Close</Button>
+      </DialogTrigger>
+      <Button>Action</Button>
+    </DialogActions>
+  </DialogContent>
+</Dialog>
+```
+
+```html
+<button aria-haspopup="true" class="fui-button">Open Dialog</button>
+<!-- ... portal ... -->
+<div aria-hidden="true" class="fui-dialog-overlay"></div>
+<div
+  aria-describedby="fui-dialog-body-id"
+  aria-labelledby="fui-dialog-title-id"
+  aria-modal="true"
+  role="alertdialog"
+  class="fui-dialog-content"
+>
+  <div id="fui-dialog-title-id" class="fui-dialog-title">
+    <span>Title</span>
+    <!-- closeButton -->
+  </div>
+  <div id="fui-dialog-body-id" class="fui-dialog-body">This is going to be inside the dialog</div>
+  <div class="fui-dialog-actions">
+    <button class="fui-button">Close</button>
+    <button class="fui-button">Action</button>
+  </div>
+</div>
+<!-- ... portal ... -->
+```
+
+### Controlled Dialog
+
+```tsx
+const CustomDialog = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const handleOpenChange = (ev, { open }) => setIsOpen(open);
+  const handleOpen = () => setIsOpen(true);
+  return (
+    <>
+      {/*
+        the trigger component is still useful outside of the Dialog,
+        to provide ARIA attributes, but it will no longer speak with the dialog.
+        A controlled state is required in this case
+      */}
+      <DialogTrigger>
+        <Button onClick={handleOpen}>Button outside Dialog Context</Button>
+      </DialogTrigger>
+      <Dialog open={isOpen} onOpenChange={handleOpenChange}>
+        <DialogContent>
+          <DialogTitle>This is an alert</DialogTitle>
+          <DialogBody>This is going to be inside the dialog</DialogBody>
+          <DialogActions>
+            {/*
+              In this case the trigger can be used to request close through `onOpenChange`,
+              as it's inside Dialog context
+            */}
+            <DialogTrigger type="close">
+              <Button>Close</Button>
+            </DialogTrigger>
+            <Button>Action</Button>
+          </DialogActions>
+        </DialogContent>
+      </Dialog>
+    </>
+  );
+};
+```
+
+```html
+<button aria-haspopup="true" class="fui-button">Open Dialog</button>
+<!-- ... portal ... -->
+<div aria-hidden="true" class="fui-dialog-overlay"></div>
+<div
+  aria-describedby="fui-dialog-body-id"
+  aria-labelledby="fui-dialog-title-id"
+  aria-modal="true"
+  role="dialog"
+  class="fui-dialog-content"
+>
+  <div id="fui-dialog-title-id" class="fui-dialog-title">
+    <span>Title</span>
+    <!-- closeButton -->
+  </div>
+  <div id="fui-dialog-body-id" class="fui-dialog-body">This is going to be inside the dialog</div>
+  <div class="fui-dialog-actions">
+    <button class="fui-button">Close</button>
+    <button class="fui-button">Action</button>
+  </div>
+</div>
+<!-- ... portal ... -->
+```
+
+### Async Input submission dialog
```

Comment:
Honestly, this looks like a usage example rather than something that needs to stay in the spec.